### ISSUE TITLE
no-unused-variable incorrectly yields error when import is used solely for type argument

### ISSUE THREAD

**Issue by compeek** [palantir/tslint#2508]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import { Component, ElementRef, EventEmitter, Input, Output, QueryList, Renderer, ViewChildren } from '@angular/core';
...
export class SomeComponent {
  @ViewChildren('example') private example: QueryList<ElementRef>;
...
```

with `tslint.json` configuration:

```json
"no-unused-variable": true
```

#### Actual behavior
ERROR: some.component.ts[1, 21]: 'ElementRef' is declared but never used.
#### Expected behavior
There should be no error since ElementRef is used as a type argument.

**Comment by andy-hanson** [comment]
Dup of #2470

**Comment by compeek** [comment]
Sorry, should have searched more thoroughly!
